src/modules/CRUD/services/MapProvider.js

```diff
@@ -29,11 +33,95 @@ export const MapProvider = ({ children }) => {
   const detailsRef = useRef(null);
   const dataTableRef = useRef(null);
 
-  const [controls, setControls] = useState([...DEFAULT_CONTROLS, ...CONTROL_LIST]);
   const [map, setMap] = useState(null);
+
+  const { i18n } = useTranslation();
+
+  const setFitBounds = useCallback(({ coordinates, hasDetails }) => {
+    if (!coordinates.length || !map) return;
+
+    const { current: detail } = detailsRef;
+    const { current: dataTable } = dataTableRef;
+
+    const padding = {
+      top: 20,
+      right: hasDetails ? (detail?.offsetWidth ?? 0 + 50) : 50,
+      bottom: !hasDetails ? (dataTable?.offsetHeight ?? 0 + 20) : 20,
+      left: 20,
+    };
+    map.resize();
+    map.fitBounds(getBounds(coordinates), { padding, duration: 0 });
+  }, [dataTableRef, detailsRef, map]);
+
+  const search = useMemo(() => ({
```

Comment:
Why a useMemo instead of two useCallback ?

Reply:
Because it's not a function but a constante

Reply:
It's an object with two function properties that are spread in a use effect if I read well ?

Reply:
Oh i misread 'two'.
I think i coded in this way because at the beginning it was easier.
But now i see what you mean :D.

Do you think it's better to use two useCallack for each method?

Reply:
Fixed 👍